src/codegen/llvm.rs

```diff
@@ -207,30 +254,52 @@ impl LLVMValue for ir::Variable {
 
 impl LLVMValue for ir::Type {
     fn llvm_represent(&self) -> String {
-        use ir::Type;
-        let mut s = String::new();
+        use ir::Type::*;
         match self {
-            Type::Void => s.push_str("void"),
-            Type::Float(n) => s.push_str(format!("f{}", n).as_str()),
-            Type::Int(n) => s.push_str(format!("i{}", n).as_str()),
-            Type::UserDefined(name) => s.push_str(format!("%{}*", name).as_str()),
+            Void => format!("void"),
+            Float(n) => format!("f{}", n),
+            Int(n) => format!("i{}", n),
+            Pointer(typ) => format!("{}*", typ.llvm_represent()),
+            Array { len, element_type } => format!("[{} x {}]", len, element_type.llvm_represent()),
+            Struct { name, .. } => format!("%{}*", name),
+            Named(name) => format!("%{}", name),
+        }
+    }
+}
+
+impl ir::Type {
+    pub(crate) fn llvm_def(&self) -> String {
+        use ir::Type::*;
+        match self {
```

Comment:
All structural type definition, but in elz there just `class`. Mapping them together is safe.